homeassistant/components/switch_as_x/__init__.py

```diff
@@ -44,12 +44,11 @@ def async_add_to_device(
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a config entry."""
-    registry = er.async_get(hass)
+    entity_registry = er.async_get(hass)
     device_registry = dr.async_get(hass)
-    try:
-        entity_id = er.async_validate_entity_id(registry, entry.options[CONF_ENTITY_ID])
-    except vol.Invalid:
-        # The entity is identified by an unknown entity registry ID
+    if (
+        entity_entry := entity_registry.async_get(entry.options[CONF_ENTITY_ID])
+    ) is None:
```

Comment:
My only concern with this check is that this in theory now limits the entities to be entities with an unique_id. I just checked with Ludeeus and apparently this is currently possible, so I am wondering what is a better approach here, as the entity might not have loaded in the state machine yet, so that is also not viable

Reply:
As discussed on Slack, it's indeed possible to set up a `switch_as_x` helper from the helpers panel tied to a switch entity not in the entity registry, so this change is not correct.